interface/Producers/ValidDiTauPairCandidatesProducers.h

```diff
@@ -58,40 +58,106 @@ class ValidDiTauPairCandidatesProducerBase: public ProducerBase<HttTypes>
 			{
 				bool diTauPairFiredTrigger = false;
-				auto trigger = product.m_detailedTriggerMatchedLeptons.at(static_cast<KLepton*>(product.m_validDiTauPairCandidates.at(0).first));
+				//std::cout << "Beginning of lambda function for " << hltNames.first << std::endl;
 				for (auto hltName: hltNames.second)
 				{
-					bool hltFired = false;
-					for (auto hlts: (*trigger))
+					if(std::find(hltPathsWithoutCommonMatch.begin(), hltPathsWithoutCommonMatch.end(), hltName) == hltPathsWithoutCommonMatch.end())
 					{
-						if (boost::regex_search(hlts.first, boost::regex(hltName, boost::regex::icase | boost::regex::extended)))
+						//std::cout << "Common match required for " << hltNames.second << std::endl;
+						// we do require a common match, check, whether both leptons are matched to trigger objects.
+						if (product.m_detailedTriggerMatchedLeptons.find(static_cast<KLepton*>(product.m_validDiTauPairCandidates.at(0).first)) != product.m_detailedTriggerMatchedLeptons.end() &&
+						    product.m_detailedTriggerMatchedLeptons.find(static_cast<KLepton*>(product.m_validDiTauPairCandidates.at(0).second)) != product.m_detailedTriggerMatchedLeptons.end())
 						{
-							for (auto matchedObjects: hlts.second)
+							auto trigger1 = product.m_detailedTriggerMatchedLeptons.at(static_cast<KLepton*>(product.m_validDiTauPairCandidates.at(0).first));
+							auto trigger2 = product.m_detailedTriggerMatchedLeptons.at(static_cast<KLepton*>(product.m_validDiTauPairCandidates.at(0).second));
+							bool hltFired1 = false;
+							bool hltFired2 = false;
+							for (auto hlts: (*trigger1))
 							{
-								if (matchedObjects.second.size() > 0) hltFired = true;
+								if (boost::regex_search(hlts.first, boost::regex(hltName, boost::regex::icase | boost::regex::extended)))
+								{
+									for (auto matchedObjects: hlts.second)
+									{
+										if (matchedObjects.second.size() > 0) hltFired1 = true;
+									}
+								}
+							}
+							//std::cout << "Found trigger for the lepton 1? " << hltFired1 << std::endl;
+							for (auto hlts: (*trigger2))
+							{
+								if (boost::regex_search(hlts.first, boost::regex(hltName, boost::regex::icase | boost::regex::extended)))
+								{
+									for (auto matchedObjects: hlts.second)
+									{
+										if (matchedObjects.second.size() > 0) hltFired2 = true;
+									}
+								}
 							}
+							//std::cout << "Found trigger for the lepton 2? " << hltFired1 << std::endl;
+							bool hltFired = hltFired1 && hltFired2;
+							// passing kinematic cuts for trigger 
+							if (lepton1LowerPtCutsByHltName.find(hltName) != lepton1LowerPtCutsByHltName.end())
+							{
+								hltFired = hltFired &&
+										(product.m_validDiTauPairCandidates.at(0).first->p4.Pt() <= *std::max_element(lepton1LowerPtCutsByHltName.at(hltName).begin(), lepton1LowerPtCutsByHltName.at(hltName).end()));
```

Comment:
In the commit you mention, the logic is fine because in case the if-statement is true, the bool for the trigger is set to false. In the case I pointed out, the opposite is implemented